overlord/state/taskrunner_test.go

```diff
@@ -184,3 +199,38 @@ func (ts *taskRunnerSuite) TestStopCancelsGoroutines(c *C) {
 
 	c.Check(invocations, Equals, 1)
 }
+
+func (ts *taskRunnerSuite) TestErrorPropagates(c *C) {
+	st := state.New(nil)
+
+	r := state.NewTaskRunner(st)
+	erroring := func(task *state.Task, tomb *tomb.Tomb) error {
+		return errors.New("boom")
```

Comment:
Curious, where is this error visible? Could we make it so that Settle would return this error (or the first error in general)?